File: src/day4/day4.py
class Solution:
    #Function to rotate an array by d elements in counter-clockwise direction. 
    def rotateArr(self, arr, d):
        n = len(arr)
        d %= n
        self.reverse(arr, 0, d - 1)
        self.reverse(arr, d, n - 1)
        self.reverse(arr, 0, n - 1)
    def reverse(self, arr, start, end):
        while start < end:
            arr[start], arr[end] = arr[end], arr[start]
            start += 1
            end -= 1
li = [1,2,3,4,5]
Answer = Solution()
Answer.rotateArr(li, 2)
print(li)

File: src/day4/day4.js
class Solution {
    // Function to rotate an array by d elements in counter-clockwise direction. 
    rotateArr(arr, d) {
        const n = arr.length
        d %= n
        this.reverse(arr, 0, d - 1)
        this.reverse(arr, d, n - 1)
        this.reverse(arr, 0, n - 1)
    }
    reverse(arr, start, end) {
        while (start < end) {
            [arr[start], arr[end]] = [arr[end], arr[start]]
            start++
            end--
        }
    }
}

const li = [1, 2, 3, 4, 5]
const Answer = new Solution()
console.log(li)
const then = new Date()
Answer.rotateArr(li, 5)
console.log(li)
console.log(`Time: ${new Date() - then} ms`)